### PR TITLE
TEC-4131 subscribe to calendar integration

### PR DESCRIPTION
**Pull request by Camwyn**
Integration of (and improvement on) the subscribe-to-calendar extension.

More performant: Removed js dependency - it can all be handled via CSS.
More modular: Split up into smaller objects and smaller templates.

Requires/required by: https://github.com/the-events-calendar/events-pro/pull/1859 _approved_

[TEC-4131]

[TEC-4131]: https://theeventscalendar.atlassian.net/browse/TEC-4131?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tribe/Views/V2/iCalendar/Links/Abstract_Link.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+		// Clean query params to only contain canonical arguments.
+		$canonical_args = [ 'post_type', 'tribe_events_cat' ];
+
+		foreach ( $view_url_args as $arg => $value ) {
+			if ( ! in_array( $arg, $canonical_args, true ) ) {
+				unset( $view_url_args[ $arg ] );
+			}
+		}
+
+		$view_url_args['tribe-bar-date'] = date( 'Y-m-d' ); // Subscribe from today.
```
Comment (bordoni) [inline src/Tribe/Views/V2/iCalendar/Links/Abstract_Link.php:101]: `build_date_object` From Date Utils.

File: src/Tribe/Views/V2/iCalendar/Links/Google_Calendar.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ */
+class Google_Calendar extends Abstract_Link {
+	/**
+	 * {@inheritDoc}
+	 */
+	public function add_subscribe_link( $template_vars, $view ) {
+		$template_vars['subscribe_links']['google'] = [
+			'display' => tribe( Service_Provider::class)->use_subscribe_links(),
+			'label' => __( 'Google Calendar', 'the-events-calendar' ),
+			'uri'   => static::get_gcal_uri( $view ),
```
Comment (bordoni) [inline src/Tribe/Views/V2/iCalendar/Links/Google_Calendar.php:29]: Indent.

File: src/Tribe/Views/V2/iCalendar/Links/Google_Calendar.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+		return $template_vars;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function add_single_subscribe_link( $links, $view ) {
+		$label = __( 'Subscribe via Google Calendar', 'the-events-calendar' );
+		$links[] = '<a class="tribe-events-gcal tribe-events-button" href="' . esc_url( static::get_gcal_uri( $view ) ) . '" title="' . esc_attr( $label ) . '">+ ' . esc_html( $label ) . '</a>';
```
Comment (bordoni) [inline src/Tribe/Views/V2/iCalendar/Links/Google_Calendar.php:40]: Indentation

File: src/Tribe/Views/V2/iCalendar/Links/Abstract_Link.php [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+ * Class Abstract_Link
+ *
+ * @package Tribe\Events\Views\V2\iCalendar
+ * @since TBD
+ */
+abstract class Abstract_Link {
+	public function register() {
+		add_filter( 'tec_views_v2_subscribe_links', [ $this, 'add_subscribe_link'], 10, 2 );
+		add_filter( 'tec_views_v2_single_subscribe_links', [ $this, 'add_single_subscribe_link' ], 10, 2 );
+	}
```
Comment (bordoni) [inline src/Tribe/Views/V2/iCalendar/Links/Abstract_Link.php:21]: Docblock.

File: src/Tribe/Views/V2/iCalendar/Links/Abstract_Link.php [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+namespace Tribe\Events\Views\V2\iCalendar\Links;
+
+/**
+ * Class Abstract_Link
+ *
+ * @package Tribe\Events\Views\V2\iCalendar
+ * @since TBD
+ */
+abstract class Abstract_Link {
```
Comment (bordoni) [inline src/Tribe/Views/V2/iCalendar/Links/Abstract_Link.php:17]: Lets create an interface for this too.

**Review by bordoni — CHANGES_REQUESTED** [review]

**Comment by Camwyn** [comment]
Current tests are passing  - but they won't tomorrow. The snapshots contain dates.

**Review by bordoni — APPROVED** [review]

**Pull request merged**